MaxText/configs/base.yml

```diff
@@ -149,9 +149,13 @@ pipeline_delay_activation_forwarding: False # This delays the activation forward
 # the communication and compute in each iteration are now independent. However this comes at the cost of doubling the pipeline bubble,
 # and you must set the number of microbatches to at least 2 * num_stages (the minimum 2 * num_stages is set by default with this delay).
 
-# Choose 'remat_policy' between 'minimal', 'save_dot_except_mlpwi', 'save_dot_except_mlp', 'save_qkv_proj', 'qkv_proj_offloaded', 'minimal_offloaded', 'save_out_proj' and 'full'.
+# Choose 'remat_policy' between 'minimal', 'save_dot_except_mlpwi', 'save_dot_except_mlp', 'save_qkv_proj', 'qkv_proj_offloaded', 'custom_save_offload' 'minimal_offloaded', 'save_out_proj' and 'full'.
 # These options offer a trade-off between speed (fastest to slowest) and HBM usage (highest to lowest)
 remat_policy: 'full'
+# If custom_save_offload remat_policy is chosen, you can select tensors from the following list to offload or keep on HBM.
+# Tensors: [decoder_layer_input,mlpwi_0,mlpwi_1,mlpwo,query_proj,key_proj,value_proj,out_proj]
+tensors_to_save: ''
```

Comment:
I'm not a huge fan of this API - I would still prefer

decoder_layer_input: '' #device, host, or remat
mlpwi_0: '' # same
mlpwi_1: '' #same

Pros: no comma separated list, easy to define sweeping runs through config combinations, less magic strings user types which could lead to errors
Cons: more config fields, requires some parsing (probably need new function in maxtext_utils that takes as input the remat config options and outputs the remat policy)

Actually having some validator that the strings user types for either approach matches valid options
E.g. in your approach validate that "decoder_layer_input,mlpwi_0" is valid vs "decder_lyr_input,mlpwif" is a typo
or in my suggested appraoch that "device", "host", and "remat" are no typoed

Reply:
Thanks for pointing out the pros and cons. Changed the API